File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.java
class Solution {
    public int countNicePairs(int[] nums) {
        long total = 0;
        int MOD = (int)Math.pow(10,9) + 7;
        HashMap<Integer,Long> map = new HashMap<Integer,Long>();
        for (int i = 0; i < nums.length; i++) map.put(nums[i]-reverseNumber(nums[i]),map.getOrDefault(nums[i]-reverseNumber(nums[i]),(long)0)+1);
        for(Map.Entry<Integer,Long> entry : map.entrySet()){
            total += entry.getValue()*(entry.getValue()-1)/2;
            total = total%MOD;
        }
        return (int)total;
    }
    public int reverseNumber(int num){
        int reversedNumber = 0;
        while(num!=0){
            reversedNumber = reversedNumber * 10 + num % 10;
            num = num/10;
        }
        return reversedNumber;
    }
}

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var reverseNumber = function (num){
    let reversedNumber = 0;
    while(num!=0){
        reversedNumber = reversedNumber * 10 + num%10;
        num = parseInt(num/10);
    }
    return reversedNumber;
}
var countNicePairs = function(nums) {
    var total = 0;
    const MOD = Math.pow(10,9) + 7;
    let map = new Map();
    for(let num of nums){
        a = num - reverseNumber(num);
        map.set(a,(map.get(a) || 0 ) + 1)
    }
    for(let [key,value] of map){
        total = (total + value*(value-1)/2)%MOD;
    }
    return total    
};